birdset/modules/models/ssast.py

```diff
@@ -0,0 +1,462 @@
+import torch.nn as nn
+import torch
+from timm.models.layers import trunc_normal_
+import timm
+import numpy as np
+from timm.models.layers import to_2tuple
+from random import randrange
+import random
+from typing import Tuple
+
+# override the timm package to relax the input shape constraint.
+class PatchEmbed(nn.Module):
+    """ Image to Patch Embedding
+    """
+    def __init__(self, img_size=224, patch_size=16, in_chans=3, embed_dim=768):
+        super().__init__()
+        img_size = to_2tuple(img_size)
+        patch_size = to_2tuple(patch_size)
+        num_patches = (img_size[1] // patch_size[1]) * (img_size[0] // patch_size[0])
+        self.img_size = img_size
+        self.patch_size = patch_size
+        self.num_patches = num_patches
+
+        self.proj = nn.Conv2d(in_chans, embed_dim, kernel_size=patch_size, stride=patch_size)
+
+    def forward(self, x):
+        x = self.proj(x).flatten(2).transpose(1, 2)
+        return x
+
+class ASTModel(nn.Module):
+    def __init__(self, label_dim=527,
+                 fshape=16, tshape=16, fstride=128, tstride=2,
+                 input_fdim=128, input_tdim=1024, model_size='base',
+                 pretrain_stage=True, load_pretrained_mdl_path=None):
+
+        super(ASTModel, self).__init__()
+        assert timm.__version__ == '0.4.5', 'Please use timm == 0.4.5, the code might not be compatible with newer versions.'
+
+        # override timm input shape restriction
+        timm.models.vision_transformer.PatchEmbed = PatchEmbed
+
+        # pretrain the AST models
+        if pretrain_stage == True:
```

Comment:
@XgamerTV Do you plan to pretrain the model?

Reply:
No, that part is needed to init the backbone that is then used for finetuning